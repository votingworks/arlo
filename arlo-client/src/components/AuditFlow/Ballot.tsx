import React, { useState, useEffect } from 'react'
import { H1, H3, Callout, H4, Divider, Button } from '@blueprintjs/core'
import styled from 'styled-components'
import { Redirect } from 'react-router-dom'
import BallotAudit from './BallotAudit'
import BallotReview from './BallotReview'
import { IAuditBoard, IReview } from '../../types'
import { BallotRow } from './Atoms'

const TopH1 = styled(H1)`
  margin: 40px 0 25px 0;
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const MainCallout = styled(Callout)`
  background-color: #137cbd;
  width: 400px;
  color: #f5f8fa;
  font-weight: 700;

  @media (max-width: 775px) {
    width: 100%;
  }
`

interface IProps {
  home: string
  roundId: string
  ballotId: string
  board: IAuditBoard
  contest: string
  previousBallot: () => void
  nextBallot: () => void
}

const Ballot: React.FC<IProps> = ({
  home,
  roundId,
  ballotId,
  board,
  contest,
  previousBallot,
  nextBallot,
}: IProps) => {
  const [auditing, setAuditing] = useState(true)
  const [review, setReview] = useState<IReview>({ vote: null, comment: '' })

  const ballot = board.ballots
    ? board.ballots.find(b => b.id === ballotId)
    : null
  useEffect(() => {
    if (ballot) {
      const { vote, comment } = ballot
      setReview({ vote, comment })
    }
  }, [ballot])

  return !board.ballots || !ballot ? (
    <Redirect to={home} />
  ) : (
    <Wrapper>
      <TopH1>{board.name}: Ballot Card Data Entry</TopH1>
      <H3>Enter Ballot Information</H3>
      <MainCallout icon={null}>
        Round {roundId}: auditing ballot {ballotId} of {board.ballots.length}
      </MainCallout>
      <BallotRow>
        <div className="ballot-side">
          <H4>Current ballot:</H4>
          <div>Tabulator: {ballot.tabulator}</div>
          <div>Batch: {ballot.batch}</div>
          <div>Record/Position: {ballot.position}</div>
        </div>
        <Divider />
        <div className="ballot-main">
          <H4>Are you looking at the correct ballot?</H4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat. Duis aute irure dolor in
            reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
            pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
            culpa qui officia deserunt mollit anim id est laborum.
          </p>
          <Button onClick={nextBallot} intent="danger">
            Ballot {ballotId} not found - move to next ballot
          </Button>
        </div>
      </BallotRow>
      {auditing ? (
        <BallotAudit
          contest={contest}
          review={review}
          setReview={setReview}
          previousBallot={previousBallot}
          goReview={() => setAuditing(false)}
        />
      ) : (
        <BallotReview
          contest={contest}
          review={review}
          goAudit={() => setAuditing(true)}
          nextBallot={nextBallot}
          previousBallot={() => {
            setAuditing(true)
            previousBallot()
          }}
        />
      )}
    </Wrapper>
  )
}

export default Ballot
