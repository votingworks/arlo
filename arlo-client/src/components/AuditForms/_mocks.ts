import { Audit } from '../../types'

export const mockAudit: Audit = {
  name: 'Primary 2019',
  riskLimit: 10,
  randomSeed: 123123123,

  contests: [
    {
      id: 'contest-1',
      name: 'Contest 1',

      choices: [
        {
          id: 'candidate-1',
          name: 'Candidate 1',
          numVotes: 42,
        },
        {
          id: 'candidate-2',
          name: 'Candidate 2',
          numVotes: 72,
        },
      ],

      totalBallotsCast: 4200,
    },
  ],

  jurisdictions: [
    {
      id: 'adams-county',
      name: 'Adams County',
      contests: ['contest-1'],
      auditBoards: [
        {
          id: 'audit-board-1',
          name: 'Audit Board #1',
          members: [],
        },
        {
          id: 'audit-board-2',
          name: 'Audit Board #2',
          members: [],
        },
      ],
      ballotManifest: {
        filename: 'Adams_County_Manifest.csv',
        numBallots: 123456,
        numBatches: 560,
        uploadedAt: '2019-06-17 11:45:00',
      },
    },
  ],

  rounds: [
    {
      startedAt: '2019-06-17 11:45:00',
      endedAt: '2019-06-17 11:55:00',
      contests: [
        {
          id: 'contest-1',
          endMeasurements: {
            pvalue: 0.085,
            isComplete: false,
          },
          results: {
            'candidate-1': 55,
            'candidate-2': 35,
          },
          sampleSize: 25,
        },
      ],
      jurisdictions: {
        'adams-county': {
          numBallots: 15,
        },
      },
    },
  ],
}

export const statusStates: Audit[] = [
  {
    name: '',
    riskLimit: '',
    randomSeed: '',
    contests: [],
    jurisdictions: [],
    rounds: [],
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [],
    rounds: [],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [
      {
        auditBoards: [
          {
            id: 'audit-board-1',
            name: 'Audit Board #1',
            members: [],
          },
        ],
        ballotManifest: {
          filename: null,
          numBallots: null,
          numBatches: null,
          uploadedAt: null,
        },
        contests: ['contest-1'],
        id: 'jurisdiction-1',
        name: 'Jurisdiction 1',
      },
    ],
    rounds: [],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [
      {
        auditBoards: [
          {
            id: 'audit-board-1',
            name: 'Audit Board #1',
            members: [],
          },
        ],
        ballotManifest: {
          filename: 'Ballot Manifest May 2019 Election - WYNADOTTE.csv',
          numBallots: 2117,
          numBatches: 10,
          uploadedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
        },
        contests: ['contest-1'],
        id: 'jurisdiction-1',
        name: 'Jurisdiction 1',
      },
    ],
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: null,
              pvalue: null,
            },
            id: 'contest-1',
            results: {},
            sampleSize: 379,
          },
        ],
        endedAt: null,
        startedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
      },
    ],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [
      {
        auditBoards: [
          {
            id: 'audit-board-1',
            name: 'Audit Board #1',
            members: [],
          },
        ],
        ballotManifest: {
          filename: 'Ballot Manifest May 2019 Election - WYNADOTTE.csv',
          numBallots: 2117,
          numBatches: 10,
          uploadedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
        },
        contests: ['contest-1'],
        id: 'jurisdiction-1',
        name: 'Jurisdiction 1',
      },
    ],
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: true,
              pvalue: 0.00020431431380638307,
            },
            id: 'contest-1',
            results: {
              'choice-1': 100,
              'choice-2': 167,
            },
            sampleSize: 379,
          },
        ],
        endedAt: 'Thu, 18 Jul 2019 16:59:34 GMT',
        startedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
      },
    ],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
]

export default mockAudit
