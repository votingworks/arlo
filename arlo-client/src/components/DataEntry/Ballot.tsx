import React, { useState, useEffect } from 'react'
import { H1, H3, Callout, H4, Button } from '@blueprintjs/core'
import styled from 'styled-components'
import { Redirect, Link } from 'react-router-dom'
import BallotAudit from './BallotAudit'
import BallotReview from './BallotReview'
import { IBallotInterpretation, IBallot, IContest } from '../../types'
import { BallotRow, FlushDivider } from './Atoms'

const TopH1 = styled(H1)`
  margin: 40px 0 25px 0;
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const MainCallout = styled(Callout)`
  background-color: #137cbd;
  width: 400px;
  color: #f5f8fa;
  font-weight: 700;

  @media (max-width: 775px) {
    width: 100%;
  }
`

interface IProps {
  home: string
  boardName: string
  roundIx: string
  batchId: string
  ballotId: number
  ballots: IBallot[]
  contest: IContest
  previousBallot: () => void
  nextBallot: () => void
  submitBallot: (
    round: string,
    batch: string,
    position: number,
    interpretation: IBallotInterpretation
  ) => void
}

const emptyInterpretation = (contestId: string) => ({
  contestId,
  interpretation: null,
  choiceId: null,
  comment: null,
})

const Ballot: React.FC<IProps> = ({
  home,
  boardName,
  roundIx,
  batchId,
  ballotId,
  ballots,
  contest,
  previousBallot,
  nextBallot,
  submitBallot,
}: IProps) => {
  const [auditing, setAuditing] = useState(true)
  const [interpretation, setInterpretation] = useState<IBallotInterpretation>(
    emptyInterpretation(contest.id)
  )

  const ballotIx = ballots
    ? ballots.findIndex(
        b => b.position === ballotId && b.batch.id === batchId
      ) /* istanbul ignore next */
    : // not showing in coverage, but is tested
      -1
  const ballot = ballots[ballotIx]

  useEffect(() => {
    if (ballot) {
      const ballotInterpretation = ballot.interpretations.find(
        i => i.contestId === contest.id
      )
      setInterpretation(ballotInterpretation || emptyInterpretation(contest.id))
    }
  }, [ballot, contest.id])

  return !ballots || !ballot || ballotIx < 0 ? (
    <Redirect to={home} />
  ) : (
    <Wrapper>
      <TopH1>{boardName}: Ballot Card Data Entry</TopH1>
      <H3>Enter Ballot Information</H3>
      <MainCallout icon={null}>
        Round {roundIx}: auditing ballot {ballotIx + 1} of {ballots.length}
      </MainCallout>
      <BallotRow>
        <div className="ballot-side">
          <H4>Current ballot:</H4>
          <div>Tabulator: {ballot.batch.tabulator}</div>
          <div>Batch: {ballot.batch.name}</div>
          <div>Record/Position: {ballot.position}</div>
        </div>
        <FlushDivider />
        <div className="ballot-main">
          <H4>Are you looking at the correct ballot?</H4>
          <p>
            Before continuing, check the &quot;Current ballot&quot; information
            to make sure you are entering data for the correct ballot. If the
            ballot could not be found, click &quot;Ballot not found&quot; below
            and move on to the next ballot.
          </p>
          <p>
            <Button onClick={nextBallot} intent="danger">
              Ballot {ballotId} not found - move to next ballot
            </Button>
          </p>
          <p>
            <Link to={home} className="bp3-button bp3-intent-primary">
              Return to audit overview
            </Link>
          </p>
        </div>
      </BallotRow>
      {auditing ? (
        <BallotAudit
          contest={contest}
          interpretation={interpretation}
          setInterpretation={setInterpretation}
          previousBallot={previousBallot}
          goReview={() => setAuditing(false)}
        />
      ) : (
        <BallotReview
          contest={contest}
          interpretation={interpretation}
          goAudit={() => setAuditing(true)}
          nextBallot={nextBallot}
          previousBallot={() => {
            setAuditing(true)
            previousBallot()
          }}
          submitBallot={(ballotInterpretation: IBallotInterpretation) =>
            submitBallot(
              roundIx,
              batchId,
              ballot.position,
              ballotInterpretation
            )
          }
        />
      )}
    </Wrapper>
  )
}

export default Ballot
