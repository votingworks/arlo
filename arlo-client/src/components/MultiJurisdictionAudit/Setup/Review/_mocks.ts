export const settingsMock = {
  empty: [
    {
      state: null,
      electionName: null,
      online: null,
      randomSeed: null,
      riskLimit: null,
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
  full: [
    {
      state: 'AL',
      electionName: 'Election Name',
      online: true,
      randomSeed: '12345',
      riskLimit: 10,
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
}

export const sampleSizeMock = {
  sampleSizes: [
    { prob: 0.54, size: 46, type: 'ASN' },
    { prob: 0.7, size: 67, type: null },
    { prob: null, size: 88, type: null },
    { prob: 0.9, size: 125, type: null },
  ],
}

export default settingsMock
