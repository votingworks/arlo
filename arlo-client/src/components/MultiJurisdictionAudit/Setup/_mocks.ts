import { ElementType } from '../../../types'
import { setupStages } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevTitle = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    title: prevTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextTitle = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    title: nextTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    title: s,
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
