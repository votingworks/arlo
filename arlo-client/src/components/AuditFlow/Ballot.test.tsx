import React from 'react'
import { render, fireEvent, wait, waitForElement } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import Ballot from './Ballot'
import { dummyBoard } from './_mocks'

const history = createMemoryHistory()

describe('Ballot', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Ballot
        home="/election/1/board/1"
        roundId="1"
        ballotId="1"
        board={dummyBoard[2]}
        contest="contest name"
        previousBallot={jest.fn()}
        nextBallot={jest.fn()}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('switches audit and review views', async () => {
    const { container, getByText, getByTestId } = render(
      <Ballot
        home="/election/1/board/1"
        roundId="1"
        ballotId="1"
        board={dummyBoard[2]}
        contest="contest name"
        previousBallot={jest.fn()}
        nextBallot={jest.fn()}
      />
    )

    fireEvent.click(getByTestId('YES'), { bubbles: true })
    await wait(() =>
      fireEvent.click(getByTestId('enabled-review'), { bubbles: true })
    )
    await wait(() => {
      expect(getByText('Submit & Next Ballot')).toBeTruthy()
      expect(container).toMatchSnapshot()
    })
    fireEvent.click(getByText('Edit'), { bubbles: true })
    await wait(() => {
      expect(getByText('Review')).toBeTruthy()
    })
  })

  it('toggles and submits comment', async () => {
    const { container, getByText, getByTestId } = render(
      <Ballot
        home="/election/1/board/1"
        roundId="1"
        ballotId="1"
        board={dummyBoard[2]}
        contest="contest name"
        previousBallot={jest.fn()}
        nextBallot={jest.fn()}
      />
    )

    fireEvent.click(getByText('Add comment'), { bubbles: true })
    await wait(() => {
      const commentInput = getByTestId('comment-textarea')
      fireEvent.change(commentInput, { target: { value: 'a test comment' } })
    })

    fireEvent.click(getByTestId('YES'), { bubbles: true })
    await wait(() =>
      fireEvent.click(getByTestId('enabled-review'), { bubbles: true })
    )
    await wait(() => {
      expect(getByText('Submit & Next Ballot')).toBeTruthy()
      expect(getByText('COMMENT: a test comment')).toBeTruthy()
      expect(container).toMatchSnapshot()
    })
  })

  it('submits review and progresses to next ballot', async () => {
    const nextBallotMock = jest.fn()
    const { container, getByText, getByTestId } = render(
      <Ballot
        home="/election/1/board/1"
        roundId="1"
        ballotId="1"
        board={dummyBoard[2]}
        contest="contest name"
        previousBallot={jest.fn()}
        nextBallot={nextBallotMock}
      />
    )

    fireEvent.click(getByTestId('YES'), { bubbles: true })

    const reviewButton = await waitForElement(
      () => getByTestId('enabled-review'),
      { container }
    )
    fireEvent.click(reviewButton, { bubbles: true })
    const nextButton = await waitForElement(
      () => getByText('Submit & Next Ballot'),
      { container }
    )
    fireEvent.click(nextButton, { bubbles: true })

    await wait(() => {
      expect(nextBallotMock).toBeCalled()
    })
  })

  it('navigates to previous ballot', async () => {
    const previousBallotMock = jest.fn()
    const { getByText, getByTestId } = render(
      <Ballot
        home="/election/1/board/1"
        roundId="1"
        ballotId="2"
        board={dummyBoard[2]}
        contest="contest name"
        previousBallot={previousBallotMock}
        nextBallot={jest.fn()}
      />
    )
    fireEvent.click(getByText('Back'), { bubbles: true })

    await wait(() => {
      expect(previousBallotMock).toBeCalledTimes(1)
    })

    fireEvent.click(getByTestId('YES'), { bubbles: true })
    await wait(() =>
      fireEvent.click(getByTestId('enabled-review'), { bubbles: true })
    )
    await wait(() => {
      expect(getByText('Submit & Next Ballot')).toBeTruthy()
    })
    fireEvent.click(getByText('Back'), { bubbles: true })
    await wait(() => {
      expect(previousBallotMock).toBeCalledTimes(2)
    })
  })

  it('redirects if ballot does not exist', async () => {
    const { container } = render(
      <Router history={history}>
        <Ballot
          home="/election/1/board/1"
          roundId="1"
          ballotId="6"
          board={dummyBoard[2]}
          contest="contest name"
          previousBallot={jest.fn()}
          nextBallot={jest.fn()}
        />
      </Router>
    )

    await wait(() => {
      expect(container).toMatchSnapshot()
    })
  })

  it('redirects if ballots array does not exist', async () => {
    const { container } = render(
      <Router history={history}>
        <Ballot
          home="/election/1/board/1"
          roundId="1"
          ballotId="1"
          board={dummyBoard[1]}
          contest="contest name"
          previousBallot={jest.fn()}
          nextBallot={jest.fn()}
        />
      </Router>
    )

    await wait(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
