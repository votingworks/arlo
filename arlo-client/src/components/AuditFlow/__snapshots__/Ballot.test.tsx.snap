// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ballot redirects if ballot does not exist 1`] = `<div />`;

exports[`Ballot redirects if ballots array does not exist 1`] = `<div />`;

exports[`Ballot renders correctly 1`] = `
<div>
  <div
    class="sc-jTzLTM iNxaQE"
  >
    <h1
      class="bp3-heading sc-VigVT cWADhG"
    >
      Audit Board #1
      : Ballot Card Data Entry
    </h1>
    <h3
      class="bp3-heading"
    >
      Enter Ballot Information
    </h3>
    <div
      class="bp3-callout sc-fjdhpX dwcRBj"
    >
      Round 
      1
      : auditing ballot 
      1
       of 
      5
    </div>
    <div
      class="sc-bdVaJa kMtymi"
    >
      <div
        class="ballot-side"
      >
        <h4
          class="bp3-heading"
        >
          Current ballot:
        </h4>
        <div>
          Tabulator: 
          11
        </div>
        <div>
          Batch: 
          Precinct 13
        </div>
        <div>
          Record/Position: 
          313
        </div>
      </div>
      <div
        class="bp3-divider"
      />
      <div
        class="ballot-main"
      >
        <h4
          class="bp3-heading"
        >
          Are you looking at the correct ballot?
        </h4>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
        <button
          class="bp3-button bp3-intent-danger"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Ballot 
            1
             not found - move to next ballot
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-bdVaJa kMtymi"
    >
      <div
        class="ballot-side"
      />
      <div
        class="ballot-main"
      >
        <h4
          class="bp3-heading"
        >
          Are you looking at the right ballot?
        </h4>
        <p>
          Step 1: Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </p>
        <p>
          Step 2: Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </p>
        <form
          class="sc-bwzfXH imPamb"
        >
          <h3
            class="bp3-heading"
          >
            contest name
          </h3>
          <div
            class="bp3-divider"
          />
          <div
            class="sc-htpNat fisUI"
          >
            <label
              class="undefined bp3-control bp3-radio sc-ifAKCX csvjng"
            >
              <input
                data-testid="YES"
                name="vote"
                type="radio"
                value="YES"
              />
              <span
                class="bp3-control-indicator"
              >
                <span
                  class="radio-text"
                >
                  Yes/For
                </span>
              </span>
            </label>
            <label
              class="undefined bp3-control bp3-radio sc-ifAKCX csvjng"
            >
              <input
                data-testid="NO"
                name="vote"
                type="radio"
                value="NO"
              />
              <span
                class="bp3-control-indicator"
              >
                <span
                  class="radio-text"
                >
                  No/Against
                </span>
              </span>
            </label>
            <label
              class="undefined bp3-control bp3-radio sc-ifAKCX csvjng"
            >
              <input
                data-testid="NO_CONSENSUS"
                name="vote"
                type="radio"
                value="NO_CONSENSUS"
              />
              <span
                class="bp3-control-indicator"
              >
                <span
                  class="radio-text"
                >
                  No audit board consensus
                </span>
              </span>
            </label>
            <label
              class="undefined bp3-control bp3-radio sc-ifAKCX csvjng"
            >
              <input
                data-testid="NO_VOTE"
                name="vote"
                type="radio"
                value="NO_VOTE"
              />
              <span
                class="bp3-control-indicator"
              >
                <span
                  class="radio-text"
                >
                  Blank vote/no mark
                </span>
              </span>
            </label>
          </div>
          <button
            class="bp3-button bp3-minimal"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-edit"
              icon="edit"
            >
              <svg
                data-icon="edit"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  edit
                </desc>
                <path
                  d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="bp3-button-text"
            >
              Add comment
            </span>
          </button>
        </form>
        <div
          class="sc-bxivhb dewsOl"
        >
          <button
            class="bp3-button bp3-disabled bp3-intent-success"
            data-testid="disabled-review"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="bp3-button-text"
            >
              Review
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Back
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Ballot switches audit and review views 1`] = `
<div>
  <div
    class="sc-jTzLTM iNxaQE"
  >
    <h1
      class="bp3-heading sc-VigVT cWADhG"
    >
      Audit Board #1
      : Ballot Card Data Entry
    </h1>
    <h3
      class="bp3-heading"
    >
      Enter Ballot Information
    </h3>
    <div
      class="bp3-callout sc-fjdhpX dwcRBj"
    >
      Round 
      1
      : auditing ballot 
      1
       of 
      5
    </div>
    <div
      class="sc-bdVaJa kMtymi"
    >
      <div
        class="ballot-side"
      >
        <h4
          class="bp3-heading"
        >
          Current ballot:
        </h4>
        <div>
          Tabulator: 
          11
        </div>
        <div>
          Batch: 
          Precinct 13
        </div>
        <div>
          Record/Position: 
          313
        </div>
      </div>
      <div
        class="bp3-divider"
      />
      <div
        class="ballot-main"
      >
        <h4
          class="bp3-heading"
        >
          Are you looking at the correct ballot?
        </h4>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
        <button
          class="bp3-button bp3-intent-danger"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Ballot 
            1
             not found - move to next ballot
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-bdVaJa kMtymi"
    >
      <div
        class="ballot-side"
      />
      <div
        class="ballot-main"
      >
        <form
          class="sc-bwzfXH imPamb"
        >
          <h3
            class="bp3-heading"
          >
            contest name
          </h3>
          <div
            class="bp3-divider"
          />
          <div
            class="sc-gZMcBi fxQdHw"
          >
            <div
              class="bp3-button-group bp3-fill bp3-large bp3-vertical"
            >
              <button
                class="bp3-button bp3-disabled bp3-fill bp3-large bp3-intent-primary sc-EHOje fCNSzj"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Yes/For
                </span>
              </button>
              <button
                class="bp3-button"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-edit"
                  icon="edit"
                >
                  <svg
                    data-icon="edit"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      edit
                    </desc>
                    <path
                      d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  Edit
                </span>
              </button>
            </div>
          </div>
          <p>
            COMMENT: 
            
          </p>
        </form>
        <div
          class="sc-bxivhb dewsOl"
        >
          <button
            class="bp3-button bp3-intent-success"
            type="submit"
          >
            <span
              class="bp3-button-text"
            >
              Submit & Next Ballot
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Back
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Ballot toggles and submits comment 1`] = `
<div>
  <div
    class="sc-jTzLTM iNxaQE"
  >
    <h1
      class="bp3-heading sc-VigVT cWADhG"
    >
      Audit Board #1
      : Ballot Card Data Entry
    </h1>
    <h3
      class="bp3-heading"
    >
      Enter Ballot Information
    </h3>
    <div
      class="bp3-callout sc-fjdhpX dwcRBj"
    >
      Round 
      1
      : auditing ballot 
      1
       of 
      5
    </div>
    <div
      class="sc-bdVaJa kMtymi"
    >
      <div
        class="ballot-side"
      >
        <h4
          class="bp3-heading"
        >
          Current ballot:
        </h4>
        <div>
          Tabulator: 
          11
        </div>
        <div>
          Batch: 
          Precinct 13
        </div>
        <div>
          Record/Position: 
          313
        </div>
      </div>
      <div
        class="bp3-divider"
      />
      <div
        class="ballot-main"
      >
        <h4
          class="bp3-heading"
        >
          Are you looking at the correct ballot?
        </h4>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
        <button
          class="bp3-button bp3-intent-danger"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Ballot 
            1
             not found - move to next ballot
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-bdVaJa kMtymi"
    >
      <div
        class="ballot-side"
      />
      <div
        class="ballot-main"
      >
        <form
          class="sc-bwzfXH imPamb"
        >
          <h3
            class="bp3-heading"
          >
            contest name
          </h3>
          <div
            class="bp3-divider"
          />
          <div
            class="sc-gZMcBi fxQdHw"
          >
            <div
              class="bp3-button-group bp3-fill bp3-large bp3-vertical"
            >
              <button
                class="bp3-button bp3-disabled bp3-fill bp3-large bp3-intent-primary sc-EHOje fCNSzj"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Yes/For
                </span>
              </button>
              <button
                class="bp3-button"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-edit"
                  icon="edit"
                >
                  <svg
                    data-icon="edit"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      edit
                    </desc>
                    <path
                      d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  Edit
                </span>
              </button>
            </div>
          </div>
          <p>
            COMMENT: 
            a test comment
          </p>
        </form>
        <div
          class="sc-bxivhb dewsOl"
        >
          <button
            class="bp3-button bp3-intent-success"
            type="submit"
          >
            <span
              class="bp3-button-text"
            >
              Submit & Next Ballot
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Back
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
